app/controllers/bus_stops_controller.rb

```diff
@@ -8,7 +8,18 @@ def autocomplete
     render json: stops.pluck(:name).sort
   end
 
-  def by_route
+  def by_sequence
+    @route = Route.find_by number: params.require(:number)
+    if @route.present?
+      @collection = @route.bus_stops_routes.group_by(&:direction).each do |_dir, bsrs|
+        bsrs.sort_by(&:sequence)
```

Comment:
Are we confident that `each` and `sort_by` is actually doing anything?

I would expect that we would need either `each` and `sort_by!` (procedural notation) or `map` and `sort_by` (functional notation).

Is there a test for this?

Reply:
No test for this.

Reply:
This is what I'll be working on.